src/org/labkey/targetedms/query/ConflictResultsManager.java

```diff
@@ -88,67 +87,76 @@ private ConflictResultsManager() {}
-        getConflictPrecursorsSql.append("INNER JOIN ");
-        getConflictPrecursorsSql.append("targetedms.generalmolecule gm2 ");
-        getConflictPrecursorsSql.append("ON gm2.Id = pep2.Id ");
-        getConflictPrecursorsSql.append("INNER JOIN ");
-        getConflictPrecursorsSql.append("targetedms.peptidegroup pg2 ");
-        getConflictPrecursorsSql.append("ON pg2.Id = gm2.PeptideGroupId ");
-        getConflictPrecursorsSql.append("INNER JOIN ");
-        getConflictPrecursorsSql.append("targetedms.runs r2 ");
-        getConflictPrecursorsSql.append("ON r2.Id = pg2.RunId ");
-        getConflictPrecursorsSql.append(" WHERE ");
-        getConflictPrecursorsSql.append(" r.Container=? " );
-        getConflictPrecursorsSql.append(" AND r2.Container=? ");
-        getConflictPrecursorsSql.append(" AND gp.RepresentativeDataState=? ");
-        getConflictPrecursorsSql.append(" AND gp2.RepresentativeDataState=? ");
-        getConflictPrecursorsSql.add(container);
-        getConflictPrecursorsSql.add(container);
-        getConflictPrecursorsSql.add(RepresentativeDataState.Conflicted.ordinal());
-        getConflictPrecursorsSql.add(RepresentativeDataState.Representative.ordinal());
-        getConflictPrecursorsSql.append(" ORDER BY gm.Id");
-
-        ConflictPrecursor[] precursors = new SqlSelector(TargetedMSManager.getSchema(), getConflictPrecursorsSql).getArray(ConflictPrecursor.class);
-        return Arrays.asList(precursors);
+        SQLFragment sql = new SQLFragment("SELECT ? AS Peptide, ");
+        sql.add(peptide);
+        sql.append("prec.Id AS newPrecursorId, ");
+        sql.append(displaySQL);
+        sql.append(" AS MoleculeName, ");
+        sql.append("r.Id AS newPrecursorRunId, ");
+        sql.append("r.filename AS newRunFile, ");
+        sql.append("prec2.Id AS oldPrecursorId, ");
+        sql.append("r2.Id AS oldPrecursorRunId, ");
+        sql.append("r2.filename AS oldRunFile\n");
+        sql.append("FROM ");
+        sql.append("targetedms.runs r ");
+        sql.append("INNER JOIN ");
+        sql.append("targetedms.peptidegroup pg ");
+        sql.append("ON r.Id = pg.RunId ");
+        sql.append("INNER JOIN ");
+        sql.append("targetedms.generalmolecule gm ");
+        sql.append("ON pg.Id = gm.PeptideGroupId ");
+        sql.append("INNER JOIN ");
+        sql.append(moleculeTable, "m");
+        sql.append(" ON gm.Id = m.Id ");
+        sql.append("INNER JOIN ");
+        sql.append("targetedms.generalprecursor gp ");
+        sql.append("ON m.Id = gp.GeneralMoleculeId ");
+        sql.append("INNER JOIN ");
+        sql.append(precursorTable, "prec");
+        sql.append(" ON gp.Id = prec.Id ");
+        sql.append("INNER JOIN ");
+        sql.append(precursorTable, "prec2");
+        sql.append(" ON prec." + joinColumn + " = prec2." + joinColumn);
+        sql.append(" INNER JOIN targetedms.generalprecursor gp2 ");
+        sql.append("ON (gp2.Id = prec2.Id AND gp.Charge = gp2.Charge) ");
```

Comment:
I think we will need to add 'gp.mz = gp2.mz' in the join condition here.  The joinColumn ("IonFormula") in line 133 can be null.

Reply:
Done